webapp/channels/src/components/channel_header_dropdown/channel_header_dropdown_items.tsx

```diff
@@ -224,6 +225,18 @@ export default class ChannelHeaderDropdown extends React.PureComponent<Props> {
                         dialogProps={{channel}}
                         text={localizeMessage('channel_header.setConversationHeader', 'Edit Conversation Header')}
                     />
+
+                    <Menu.ItemToggleModalRedux
```

Comment:
The create channel permission is associated with specific team. Since the operation wouldn't necessarily be performed on the team the user is currently on, it would be incorrect to check for create private channel permission here. I've added create private channel permission check on API level, where we know the team where the channel is being created.